### DISCUSSION TITLE
Anyone have a functioning vanilla like stair placement rule?

### DISCUSSION THREAD

**Discussion by Tofaa2** [Minestom/Minestom#1362] [Help] [answered]
As the title says I'm looking for a working vanilla like stair placement rule

**Comment by MrGazdag** [comment, marked as answer]
You cannot create a functioning star placement rule correctly, until PlacementRules change a bit. The server sends the clicked position on the block, thus the server can calculate whether the stair is going to be a lower half (clicked Y < 0.5) or an upper half (clicked Y >= 0.5). This information is not visible to placement rules currently.
I made a PR to add it without any breaking changes (#443), but it has not been accepted yet, because they dob’t like the current version of the API and they will likely replace it with events. Work was started on it, but it hasn’t been touched in a long time.

**Reply by Tofaa2** [reply]
Thanks for the reply and the pr link. Ill take a look
